server/player/provider/leveldb.go

```diff
@@ -0,0 +1,57 @@
+package provider
+
+import (
+	"encoding/json"
+	"github.com/df-mc/dragonfly/server/player"
+	"github.com/df-mc/goleveldb/leveldb"
+	"github.com/google/uuid"
+	"os"
+)
+
+// DBProvider ...
+type DBProvider struct {
+	db *leveldb.DB
+}
+
+// NewDBProvider creates a new player data provider that saves and loads data using
+// a LevelDB database.
+func NewDBProvider(path string) (*DBProvider, error) {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		_ = os.Mkdir(path, 0777)
+	}
+	db, err := leveldb.OpenFile(path, nil)
+	if err != nil {
+		return nil, err
+	}
+	return &DBProvider{db: db}, nil
+}
+
+// Save ...
+func (p *DBProvider) Save(d player.Data) {
+	data := toJson(d)
+	jsondata, err := json.Marshal(data)
```

Comment:
Should this data really be in JSON? For blocks, their states might load incorrectly since we wouldn't know the value types

Reply:
That is indeed a problem, I wasn't really sure how else to store it. If you know a good way what do you think would be a way to store it that keeps the value types?

Reply:
NBT should store types, we already use it for creative items anyways